src/elexsolver/TransitionSolver.py

```diff
@@ -0,0 +1,111 @@
+import logging
+import warnings
+from abc import ABC
+
+import numpy as np
+
+from elexsolver.logging import initialize_logging
+
+initialize_logging()
+
+LOG = logging.getLogger(__name__)
+
+
+class TransitionSolver(ABC):
+    """
+    Abstract class for transition solvers.
+    """
+
+    def __init__(self):
+        self._transitions = None
+
+    def fit_predict(self, X: np.ndarray, Y: np.ndarray, weights: np.ndarray | None = None):
+        """
+        After this method finishes, transitions will be available in the `transitions` class member.
+
+        Parameters
+        ----------
+        `X` : np.ndarray matrix or pandas.DataFrame of int
+            Must have the same number of rows as `Y` but can have any number of columns greater than the number of rows.
+        `Y` : np.ndarray matrix or pandas.DataFrame of int
+            Must have the same number of rows as `X` but can have any number of columns greater than the number of rows.
+        `weights` : list, np.ndarray, or pandas.Series of int, optional
+            Must have the same length (number of rows) as both `X` and `Y`.
+
+        Returns
+        -------
+        np.ndarray matrix of float of shape (number of columns in `X`) x (number of columns in `Y`).
+        Each float represents the percent of how much of row x is part of column y.
+        """
+        raise NotImplementedError
+
+    def get_prediction_interval(self, pi: float):
```

Comment:
do we need this function?

Reply:
Good catch!  I'll remove it.  I didn't get that far lol 😅